tests/test_probe.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+
+import ucp
+
+
+@pytest.mark.asyncio
+@pytest.mark.skipif(
+    ucp.get_ucx_version() < (1, 11, 0),
+    reason="Endpoint error handling is unreliable in UCX releases prior to 1.11.0",
+)
+@pytest.mark.parametrize("transfer_api", ["am", "tag"])
+async def test_message_probe(transfer_api):
+    msg = bytearray(b"0" * 10)
+
+    async def server_node(ep):
+        # Wait for remote endpoint to close before probing the endpoint for
+        # in-transit message and receiving it.
+        while not ep.closed():
+            pass
```

Comment:
```suggestion
            await asyncio.sleep(0)  # Yield task
```

Reply:
For robustness, yield the asyncio task as part of the spinlock.

Reply:
Good idea, applied in https://github.com/rapidsai/ucx-py/pull/806/commits/55a9a2527d91493601862c7cc11c949267aad0e4.